tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/CaeAppTests.cs

```diff
@@ -0,0 +1,93 @@
+        private readonly string _tenantId = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+        private readonly KeyVaultSecretsProvider _keyVaultProvider = new(KeyVaultInstance.MsalTeam);
+        private readonly HttpClient _httpClient = new();
+        private readonly TimeSpan _delayTimeout = TimeSpan.FromMinutes(5);
+        private readonly string[] _graphAppScope = ["https://graph.microsoft.com/.default"];
+
+        [TestInitialize]
+        public async Task TestInitialize()
+        {
+            if (string.IsNullOrEmpty(_confidentialClientSecret))
+            {
+                _confidentialClientSecret = _keyVaultProvider.GetSecretByName(TestConstants.MsalCCAKeyVaultSecretName).Value;
+            }
+            await LabUserHelper.EnableAppServicePrincipal(LabApiConstants.LabCaeConfidentialClientId).ConfigureAwait(false);
+        }
+        [TestCleanup]
+        public async Task TestCleanup()
+        {
+            await LabUserHelper.EnableAppServicePrincipal(LabApiConstants.LabCaeConfidentialClientId).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// [Init] Enable the app's service principal.
+        /// Initialize CAE CCA and get an app token for Graph.
+        /// Call Graph's /users endpoint with the token; should receive a successful response.
+        /// Disable the CAE app's service principal and wait until the changes propagate to Graph.
+        /// Use the cached token to call Graph again; should receive a 401 with claims. 
+        /// Acquire a new token with claims; should receive an exception that the SP is disabled.
+        /// [Cleanup] Enable the app's service principal again.
+        /// </summary>
+        [TestMethod]
+        public async Task ClientCredentials_WithDisabledServicePrincipal_ThrowsException()
+        {
+            var cca = ConfidentialClientApplicationBuilder
+                .Create(_caeAppClientId)
+                .WithClientSecret(_confidentialClientSecret)
+                .WithTenantId(_tenantId)
+                .WithClientCapabilities(new[] { "cp1" })
+                .WithLegacyCacheCompatibility(false)
+                .Build();
+
+            var result = await cca.AcquireTokenForClient(_graphAppScope).ExecuteAsync().ConfigureAwait(false);           
+
+            var response1 = await _httpClient.SendAsync(CreateRequest(result)).ConfigureAwait(false);
+            Assert.AreEqual(HttpStatusCode.OK, response1.StatusCode);
+
+            await LabUserHelper.DisableAppServicePrincipal(LabApiConstants.LabCaeConfidentialClientId).ConfigureAwait(false);
+            await Task.Delay(_delayTimeout).ConfigureAwait(false);
+
+            var response2 = await _httpClient.SendAsync(CreateRequest(result)).ConfigureAwait(false);
+            Assert.AreEqual(HttpStatusCode.Unauthorized, response2.StatusCode);
+
+            var claims = WwwAuthenticateParameters.GetClaimChallengeFromResponseHeaders(response2.Headers);
+
+            Assert.IsNotNull(claims);
```

Comment:
pls add a comment here with how the claims looks like (at present)

Reply:
is it not possible to assert the content of the claim?